ProcessMaker/processmaker
# FOUR-19037 Refactor template importing

## Issue & Reproduction Steps
The Expense Report guided template was taking about 45 seconds to run locally (post to `api/1.0/template/create/process/53`)

## Solution
- Cache Schema lookups
- Refactor Options
- Remove generating manifest (expensive) from every associate categories call
- Prevent duplicate calls, short circuit re-order so expensive calls are last, other basic cleanup.

## How to Test
Sync guided templates and install the expense report (you can skip the HR system). Observe how long it takes to install the process and go the dashboard

## Related Tickets & Packages
- https://processmaker.atlassian.net/browse/FOUR-19037

ci:deploy
ci:next

## Code Review Checklist
